rotkehlchen/chain/ethereum/modules/uniswap/v2/common.py

```diff
@@ -127,11 +134,198 @@ def decode_uniswap_v2_like_swap(
+
+
+def decode_uniswap_like_deposit_and_withdrawals(
+        tx_log: EvmTxReceiptLog,
+        decoded_events: list[HistoryBaseEntry],
+        action_items: list[ActionItem],
+        all_logs: list[EvmTxReceiptLog],
+        event_action_type: Literal['addition', 'removal'],
+        counterparty: str,
+        database: 'DBHandler',
+        ethereum_inquirer: 'EthereumInquirer',
+        factory_address: ChecksumEvmAddress,
+        init_code_hash: str,
+) -> None:
+    """
+    This is a common logic for Uniswap V2 like AMMs e.g Sushiswap.
+    This method decodes a liquidity addition or removal to Uniswap V2 pool.
+
+    Examples of such transactions are:
+    https://etherscan.io/tx/0x1bab8a89a6a3f8cb127cfaf7cd58809201a4e230d0a05f9e067674749605959e (deposit)
+    https://etherscan.io/tx/0x0936a16e1d3655e832c60bed52040fd5ac0d99d03865d11225b3183dba318f43 (withdrawal)
+    """  # noqa: E501
+    resolved_eth = A_ETH.resolve_to_crypto_asset()
+    maybe_pool_address = tx_log.address
+    amount0_raw = hex_or_bytes_to_int(tx_log.data[:32])
+    amount1_raw = hex_or_bytes_to_int(tx_log.data[32:64])
+
+    token0: Optional[CryptoAsset] = None
+    token1: Optional[CryptoAsset] = None
+    event0_idx = event1_idx = None
+
+    if event_action_type == 'addition':
+        notes = 'Add {} {} of liquidity to {} LP {}'
+        from_event_type = (HistoryEventType.SPEND, HistoryEventSubType.NONE)
+        to_event_type = (HistoryEventType.DEPOSIT, HistoryEventSubType.DEPOSIT_ASSET)
+    else:
+        notes = 'Remove {} {} of liquidity from {} LP {}'
+        from_event_type = (HistoryEventType.RECEIVE, HistoryEventSubType.NONE)
+        to_event_type = (HistoryEventType.WITHDRAWAL, HistoryEventSubType.REMOVE_ASSET)
+
+    # First, get the tokens deposited into the pool. The reason for this approach is to circumvent  # noqa: E501
+    # scenarios where the mint/burn event comes before the needed transfer events.
+    for other_log in all_logs:
+        if other_log.topics[0] == TRANSFER_SIGNATURE and hex_or_bytes_to_int(other_log.data[:32]) == amount0_raw:  # noqa: E501
+            token0 = get_or_create_evm_token(
+                userdb=database,
+                evm_address=other_log.address,
+                chain_id=ChainID.ETHEREUM,
+                token_kind=EvmTokenKind.ERC20,
+                evm_inquirer=ethereum_inquirer,
+            )
+            token0 = resolved_eth if token0 == A_WETH else token0
+        elif other_log.topics[0] == TRANSFER_SIGNATURE and hex_or_bytes_to_int(other_log.data[:32]) == amount1_raw:  # noqa: E501
+            token1 = get_or_create_evm_token(
+                userdb=database,
+                evm_address=other_log.address,
+                chain_id=ChainID.ETHEREUM,
+                token_kind=EvmTokenKind.ERC20,
+                evm_inquirer=ethereum_inquirer,
+            )
+            token1 = resolved_eth if token1 == A_WETH else token1
```

Comment:
are there WETH pools? what happens if you are really depositing weth? Can you check that case?

Reply:
WETH pools? Idg.

All XX/ETH pools use WETH to represent ETH. Unless I misunderstood

Reply:
Okey true. Then I believe this should be always WETH if is ETH, thanks